### DISCUSSION TITLE
Is this a correct approach if I want to create a wrapper component around a Mantine component?

### DISCUSSION THREAD

**Discussion by abhi-simform** [mantinedev/mantine#6201] [Q&A] [answered]
I want to create a wrapper button component on top of Mantine button component. Is this the correct way to do it or should I do it some other way. It seems to work fine this way, but I am not sure if this is the best way to do it.
<img width="955" alt="image" src="https://github.com/mantinedev/mantine/assets/99975773/1259087c-3c20-497e-b738-a57489c4f4a8">

**Comment by yverby** [comment, marked as answer]
Follow this guide
https://mantine.dev/guides/polymorphic/#wrapping-polymorphic-components

**Reply by abhi-simform** [reply]
@yverby thank you so much for your response. I read that earlier but missed the second example where it shows the polymorphic wrapper.
